src/segment.go

```diff
@@ -112,16 +112,42 @@ func (segment *Segment) getValue(property Property, defaultValue string) string
 	return defaultValue
 }
 
-func (segment *Segment) shouldIgnoreFolder(cwd string) bool {
-	if value, ok := segment.Properties[IgnoreFolders]; ok {
+func (segment *Segment) shouldIncludeFolder(cwd string) bool {
+	included := true
+	if value, ok := segment.Properties[IncludeFolders]; ok {
 		list := parseStringArray(value)
-		for _, element := range list {
-			pattern := fmt.Sprintf("^%s$", element)
-			matched := matchString(pattern, cwd)
-			return matched
+		if len(list) > 0 {
+			included = cwdMatchesOneOf(cwd, list)
 		}
+	}
+
+	if !included {
```

Comment:
this is hard for me to reason about:

- if there are no values in both lists, it's true.
- if the folder is included, but not excluded it's true
- if the folder is not included and include folders has elements, it's false?
- if the folder is included and excluded, do we prioritise included?
- if the folder is excluded, it's false
- if the folder is not excluded and exclude folders has elements, it's true?

I would like to see this logic reflected in the tests as well, because now I feel we made some assumptions.

Reply:
I believe the test cases cover all of these examples.  Below I've quoted your scenarios and which test case I believe covers it. Let me know if there's anything I'm missing there.

```
if there are no values in both lists, it's true. → "Base Case"
if the folder is included, but not excluded it's true → "Included" and "Include Match / Exclude Mismatch"
if the folder is not included and include folders has elements, it's false? → "Include Mismatch"
if the folder is included and excluded, do we prioritise included? → "Include Match / Exclude Match"
if the folder is excluded, it's false → "Exclude"
if the folder is not excluded and exclude folders has elements, it's true? → "Exclude Mismatch"
```

I'll try to refactor the production code logic to make it more clear 👍.

Reply:
If it does, perfect, I gave some pointers to make it a bit more readable which imho should be good to go when the tests pass. Thanks!

Reply:
Please review the refactored code. I think that what's getting you tripped up is the different assumptions made if the IncludedFolders or ExcludedFolders are unspecified. I tried to think about the user's intention with each of these.

* If IncludedFolders is unspecified, everything is included
* If IncludedFolders is specified, the user wants the segment to be displayed only when CWD matches at least 1 included folder.
* If ExcludedFolders is unspecified, nothing is excluded
* If ExcludedFolders is specified, the user wants the segment to be displayed only when CWD matches 0 excluded folders.
* If both are specified, the user wants the segment to be displayed only when CWD matches at least 1 included folders and matches 0 ExcludedFolders 

Maybe a way to think about this logic is that it's both an Allow List and a Deny List.  I've left the changes unsquashed for now, in case you wanted to compare or anything.  If this looks good to you, I will squash them.